mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp

```diff
@@ -2515,8 +2515,12 @@ convertOmpTargetData(Operation *op, llvm::IRBuilderBase &builder,
           argIndex++;
         }
 
-        bodyGenStatus = inlineConvertOmpRegions(region, "omp.data.region",
-                                                builder, moduleTranslation);
+        SmallVector<llvm::PHINode *> phis;
+        llvm::BasicBlock *continuationBlock =
+            convertOmpOpRegions(region, "omp.data.region", builder,
+                                moduleTranslation, bodyGenStatus, &phis);
+        builder.SetInsertPoint(continuationBlock,
+                               continuationBlock->getFirstInsertionPt());
```

Comment:
Just out of curiosity: Why did the original call break things? I see the only difference is that it does some special handling if there is a single block in the region.

Reply:
The special case assumes that that no new basic blocks are created, which does happen for omp.target, so it seems to generate code in the wrong place because of this.

Reply:
We currently still have this change only in ATD, should we upstream it or has it somehow become unnecessary? I don't actually know how to reproduce the issue it fixes, that's why I ask instead of checking myself @jsjodin @TIFitis